src/Tizen.NUI.Components/Controls/NumberPicker.cs

```diff
@@ -0,0 +1,531 @@
+            pickerScroller.ScrollAnimationStarted += OnScrollAnimationStarted;
+
+            itemList = new List<TextLabel>();
+            
+            minValue = maxValue = currentValue = 0;
+            displayedValues = null;
+            //Those many flags for min, max, value method calling sequence dependency.
+            needItemUpdate = true;
+            displayedValuesUpdate = false;
+            onAnimation = false;
+            loopEnabled = false;
+
+            startScrollOffset = (int)numberPickerStyle.NumberPickerStartScrollOffset.Height;
+            itemHeight = (int)numberPickerStyle.NumberPickerItemTextLabel.Size.Height;
+            startScrollY = (itemHeight * dummyItemsForLoop) + startScrollOffset;
+            startY = startScrollOffset;
+
+            Add(pickerScroller);
+            addLine();
+        }
+
+        private void onNumberPickerValueChanged()
+        { 
+            NumberPickerValueChangedEventArgs eventArgs = 
+                new NumberPickerValueChangedEventArgs(displayedValuesUpdate ? Int32.Parse(itemList[currentValue].Name) : Int32.Parse(itemList[currentValue].Text));
+            NumberPickerValueChanged?.Invoke(this, eventArgs);
+        }
+
+        private void OnScroll(object sender, ScrollEventArgs e)
+        {
+            //Ignore if the scrolling animation happened.
+            if (!loopEnabled || onAnimation) return;
+            
+            //Check the scroll is going out to the dummys if so, bring it back to page.
+            if (e.Position.Y > -(startScrollY - (itemHeight * 2))) 
+                pickerScroller.ScrollTo(-e.Position.Y + pageSize, false);
+            else if (e.Position.Y < -(startScrollY + pageSize - (itemHeight * 2)))
+                pickerScroller.ScrollTo(-e.Position.Y - pageSize, false);
+        }
+
+        private void OnScrollAnimationStarted(object sender, ScrollEventArgs e)
+        {
+            onAnimation = true;
+        }
+
+        private void OnScrollAnimationEnded(object sender, ScrollEventArgs e)
+        {
+            //Ignore if the scroll position was not changed. (called it from this function)
+            if (lastScrollPosion == (int)e.Position.Y) return;
+
+            //Calc offset from closest item.
+            int offset = (int)(e.Position.Y + startScrollOffset) % itemHeight;
+            if (offset < -(itemHeight / 2)) offset += itemHeight;
+
+            lastScrollPosion = (int)(-e.Position.Y + offset);
+
+            //Item center align with animation, otherwise changed event emit.
+            if (offset != 0) 
+                pickerScroller.ScrollTo(-e.Position.Y + offset, true);
+            else {
+                currentValue = ((int)(-e.Position.Y / itemHeight) + 2);
+                onNumberPickerValueChanged();
+            }
+
+            onAnimation = false;
+        }
+
+        //This is UI requirement. It helps where exactly center item is.
+        private void addLine()
```

Comment:
Normally, both `public` and `private` have method name start with capital character.

Reply:
Done.